src/server/req_movejob.c

```diff
@@ -223,10 +225,18 @@ req_orderjob(struct batch_request *req)
 
 		/* Jobs are in different queues */
 
-		if ((rc = svr_chkque(pjob1, pjob2->ji_qhdr, get_jattr_str(pjob1, JOB_ATR_submit_host),
+		conn = get_conn(req->rq_conn);
+		if (conn) {
+			physhost = conn->cn_physhost;
+		}
+
+		if ((rc = svr_chkque(pjob1, pjob2->ji_qhdr,
+				     get_jattr_str(pjob1, JOB_ATR_submit_host),
+				     physhost,
 				     MOVE_TYPE_Order)) ||
 		    (rc = svr_chkque(pjob2, pjob1->ji_qhdr,
 				     get_jattr_str(pjob2, JOB_ATR_submit_host),
+				     physhost,
```

Comment:
You are setting the rc variable twice and then rejecting the request with it.  Maybe you want to set two different variables and then determine which one to send back?  I mean we are rejecting because both of these fail.  You only return the rc from the second job.

Reply:
The OR operator is used, which means the first rc != 0 is rejected. It can be rc from the first job check. I prefer to keep it as is.